src/DepositHandler.sol

```diff
@@ -126,23 +157,81 @@ contract DepositHandler is Pausable, AccessControl, IDepositHandlerErrors {
      * @notice Remove contract from pause, and allow interraction with critical functions.
      * @dev Manager is able to unpause a contract when vulnerability or
      * any other problem is resolved. Functions using `whenNotPaused()` modifier will work.
-     * Function restricted to be called only by `MANAGER` of this contract.
+     * Function restrictions:
+     *  - Can only be called by `MANAGER` of this contract.
      */
     function unpause() private onlyRole(MANAGER) {
         _unpause();
     }
 
-     /*//////////////////////////////////////////////////
-                VIEW FUNCTIONS
-    /////////////////////////////////////////////////*/
     /**
-     * @dev Allow to query a deposited amount of tokens for `_depositor`.
+     * @notice Allow a list of selected participants, who passed interview stage, 
+     * to do a deposit to the bootcamp.
+     * @dev Set `Deposit` status for all addresses in the `_participants` array.
+     * Faster way to set status for a a list of participants instead of calling on eby one.
+     * Function restrictions:
+     *  - Can only be called by `MANAGER` of this contract.
      * 
-     * @param _depositor - address of the person who did a deposit.
+     * @param _participants - array of participants addresses.
+     */
+    function allowDepositBatch(address[] calldata _participants) external onlyRole(MANAGER) {
```

Comment:
we could remove the Deposit status in the Status enum so Encode does not need to collect addresses from everyone. anyone can then deposit and once they do, the Status will change to InProgress.

Reply:
I was thinking about `Deposit` status as an additional level of protection for the bootcamp, to not allow some random address deposit funds to the contract. I understand that this is quite hard to deposit by mistake a funds, but anyways decided to add this, to allow users depositing only after manager approval.
Also, once manager set a Deposit status thorough a `allowDepositBatch` we can emit an event and grab addresses allowed for a deposit from the event and on UI show new button for depositing. But I am not sure how efficient is to emit an event with array of addresses, and I think we can handle this also from the UI part.